pkgs/tools/misc/trash-cli/default.nix

```diff
@@ -43,6 +43,16 @@ python3Packages.buildPythonApplication rec {
 
     runHook postInstallCheck
   '';
+  postInstall = ''
+    for bin in trash{,-{empty,list,put,restore}}; do
+      $out/bin/$bin --print-completion bash > $bin
```

Comment:
There is a ready made tool to do this

https://github.com/caarlos0/nixpkgs/blob/28c08a222e13bf9533e3f92fb54af08c1658a509/pkgs/tools/package-management/nfpm/default.nix#L31

Reply:
Can we iterate over all files in /bin instead of listing them manually?

Reply:
@bjornfor I had a look and iteration is not really possible since there are other commands in the bin directory that do not support the `--print-completion` option.

@lucasew Indeed, using `installShellCompletion` would be much nicer.

@Freed-Wu Could you try switching this PR over to

``` patch
diff --git a/pkgs/tools/misc/trash-cli/default.nix b/pkgs/tools/misc/trash-cli/default.nix
index f607da9099a..9a106e13ac8 100644
--- a/pkgs/tools/misc/trash-cli/default.nix
+++ b/pkgs/tools/misc/trash-cli/default.nix
@@ -1,4 +1,4 @@
-{ lib, fetchFromGitHub, python3Packages }:
+{ lib, fetchFromGitHub, installShellFiles, python3Packages }:
 
 python3Packages.buildPythonApplication rec {
   pname = "trash-cli";
@@ -11,7 +11,9 @@ python3Packages.buildPythonApplication rec {
     hash = "sha256-TJEi7HKIrfOdb+LLRt2DN5gWdFzUeo6isb59lFLK4bQ=";
   };
 
-  propagatedBuildInputs = with python3Packages; [ psutil six ];
+  nativeBuildInputs = [ installShellFiles ];
+
+  propagatedBuildInputs = with python3Packages; [ psutil six shtab ];
 
   nativeCheckInputs = with python3Packages; [
     mock
@@ -22,6 +24,14 @@ python3Packages.buildPythonApplication rec {
     sed -i '/typing/d' setup.cfg
   '';
 
+  postInstall = ''
+    for bin in trash-empty trash-list trash-restore trash-put trash; do
+      installShellCompletion --cmd "$bin" \
+        --bash <("$out/bin/$bin" --print-completion bash) \
+        --zsh  <("$out/bin/$bin" --print-completion zsh)
+    done
+  '';
+
   doInstallCheck = true;
   installCheckPhase = ''
     runHook preInstallCheck
```

instead? I'm not sure how it works for you without adding a `shtab` build input but I needed it.

Reply:
> I'm not sure how it works for you without adding a shtab build input but I needed it.

`shtab` is only used in build stage to generate shell completion scripts, and remove it will not effect `trash-cli` work normally. So is there any special reason to need it?

Reply:
Having it as a build input only during build is fine by me. But I do need it to be available even during build:

``` console
$ nix build -f. trash-cli ; head result/share/bash-completion/completions/trash.bash # Without shtab
Please install shtab firstly!

$ nix build -f. trash-cli ; head result/share/bash-completion/completions/trash.bash # With shtab
# AUTOMATICALLY GENERATED by `shtab`



_shtab_trash_option_strings=('-h' '--help' '--print-completion' '-d' '--directory' '-f' '--force' '-i' '--interactive' '-r' '-R' '--recursive' '--trash-dir' '-v' '--verbose' '--version')

_shtab_trash_pos_0_COMPGEN=_shtab_compgen_files
_shtab_trash___trash_dir_COMPGEN=_shtab_compgen_dirs

_shtab_trash___print_completion_choices=('bash' 'zsh' 'tcsh')
```